homework-4/script.js

```diff
@@ -0,0 +1,247 @@
+    let tr = document.createElement('tr');
+    let tdCompanyName = document.createElement('td');
+    let tdContactName = document.createElement('td');
+    let tdAddress = document.createElement('td');
+    let tdCity = document.createElement('td');
+    let tdCountry = document.createElement('td');
+    let checkBox = document.createElement('input');
+    let editButton = document.createElement('button');
+    let deleteButton = document.createElement('button');
+    checkBox.type='checkbox';
+    checkBox.className = 'checkbox-tr';
+    tdCompanyName.className = 'company';
+    tdContactName.className = 'contact';
+    tdAddress.className = 'address';
+    
+    deleteButton.type='button';
+    editButton.type='button';
+    deleteButton.setAttribute('data-id', element.id);
+    editButton.setAttribute('data-id', element.id);
+    editButton.setAttribute('data-name', element.name);
+    editButton.setAttribute('data-company', element.company);
+    editButton.setAttribute('data-address', element.address);
+    editButton.setAttribute('data-city', element.city);
+    editButton.setAttribute('data-country', element.country);
+    deleteButton.setAttribute("disabled", true);
+
+    checkBox.addEventListener('change', function (element) {
+      if (element.currentTarget.checked) { 
+        deleteButton.removeAttribute("disabled", false);
+      }   
+    });
+
+    editButton.addEventListener('click', (ele) => {
+      const modalEditObject = document.querySelector('.modal-edit-object');
+      modalEditObject.style.display = 'block';
+      let targetElement = ele.currentTarget;
+      let id = targetElement.getAttribute('data-id');
+
+      contactName.value = targetElement.getAttribute('data-name');
+      companyName.value = targetElement.getAttribute('data-company');
+      address.value = targetElement.getAttribute('data-address');
+      city.value = targetElement.getAttribute('data-city');
+      country.value = targetElement.getAttribute('data-country');
+
+      updateUsers.addEventListener('click', () => {
+        editElement(id);    
+      });              
+  });
+
+    tr.appendChild(checkBox);
+    tr.appendChild(tdCompanyName);
+    tr.appendChild(tdContactName);
+    tr.appendChild(tdAddress);
+    tr.appendChild(tdCity);
+    tr.appendChild(tdCountry);
+    tr.appendChild(editButton);
+    tr.appendChild(deleteButton);
+    tableBody.appendChild(tr);
+
+    tdCompanyName.innerText = element.name;
+    tdContactName.innerText = element.company;
+    tdAddress. innerText = element.address;
+    tdCity.innerText = element.city;
+    tdCountry.innerText = element.country;
+    deleteButton.innerHTML = '<i class="fa fa-close icon-close">';
+    editButton.innerHTML = '<i class="fa fa-pen icon-pen">';
+    editButton.className = 'edit';
+    deleteButton.className ='delete';
+    tdCompanyName.className ='name';
+    tr.className = 'add-tr';
+
+    // filter
+    function filterFunction(filterName, searchText) {
```

Comment:
Do you really need function declaration inside of another function?

Reply:
transferred